aten/src/ATen/native/Copy.cpp

```diff
@@ -56,8 +39,13 @@ Tensor& _s_copy__cpu(Tensor& self, const Tensor& src, bool non_blocking) {
     _s_copy_from(src, self, non_blocking);
     return self;
   }
-  AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Half, at::ScalarType::Bool,
-      self.scalar_type(), "_copy__cpu", [&]() { ::_copy__cpu<scalar_t>(self, src); });
+
+  if (self.scalar_type() == src.scalar_type()) {
+    copy_kernel_same_type(kCPU, self, src);
+  } else {
+    AT_CHECK(self.numel() == src.numel(), "sizes do not match");
```

Comment:
Numel doesn't give you all info about the size. I'm not sure what a better word might be.

Reply:
It is copy of the current aten/src/ATen/native/Copy.cpp:23 . I have no better working for it.

Reply:
Also _"sizes do not match"_ is a bad error message. We should mention what the shapes were, and what't the problem.

Reply:
This should probably be an `AT_ASSERT`, but I don't think it's super important. We do a proper shape check before this in `copy_` via `expand_inplace`.